src/kernels/execution/kernelExecution.ts

```diff
@@ -44,22 +43,20 @@ export class KernelExecution implements IDisposable {
     constructor(
         private readonly kernel: IKernel,
         appShell: IApplicationShell,
-        readonly metadata: Readonly<KernelConnectionMetadata>,
         private readonly interruptTimeout: number,
-        controller: NotebookController,
```

Comment:
`controller` can be accessed from `kernel` argument, hence we don't need to pass this as a separate argument.